agent/api/ecr.go

```diff
@@ -25,4 +31,23 @@ type ECRAuthData struct {
 	EndpointOverride string `json:"endpointOverride"`
 	Region           string `json:"region"`
 	RegistryID       string `json:"registryId"`
+	UseExecutionRole bool   `json:"useExecutionRole"`
```

Comment:
Is this persisted to the disk? If yes, you'd also need to do a version bump of the state file.